counsel.el

```diff
@@ -124,7 +124,9 @@ To skip the `executable-find' check, start the string with a space."
                      (car (split-string cmd)))))
       (or (and (stringp program)
                (not (string= program ""))
-               (executable-find program))
+               (if (<= 27 emacs-major-version)
+                   (executable-find program (file-remote-p default-directory))
```

Comment:
`executable-find` already checks `(file-remote-p default-directory)`, so AFAICT you can just pass `t` as the second argument.